### PR TITLE
gnrc/netif: convert to ztimer

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This converts gnrc_netif.c to use ztimer instead of xtimer.

### Testing procedure

I think the test in tests ` tests/gnrc_netif_ipv6_wait_for_global_address` should be sufficient:

```Shellsession
koen@zometeen ~/dev/RIOT-dev $ make -C tests/gnrc_netif_ipv6_wait_for_global_address/ term
make: Entering directory '/home/koen/dev/RIOT-dev/tests/gnrc_netif_ipv6_wait_for_global_address'
/home/koen/dev/RIOT-dev/tests/gnrc_netif_ipv6_wait_for_global_address/bin/native/tests_gnrc_netif_ipv6_wait_for_global_address.elf /dev/ttyACM0 
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

Help: Press s to start test, r to print it is ready
s
START
....
OK (4 tests)
```

### Issues/PRs references

On the list of #17111

### CODE REVIEW

File: sys/net/gnrc/Makefile.dep [hunk 1]
```diff
@@ -135,6 +135,7 @@ ifneq (,$(filter gnrc_netif,$(USEMODULE)))
   USEMODULE += netif
   USEMODULE += l2util
   USEMODULE += fmt
+  USEMODULE += ztimer_msec
```
Comment (fjmolinas) [inline sys/net/gnrc/Makefile.dep:138]:
```suggestion
  USEMODULE += ztimer_msec
  USEMODULE += ztimer_usec
```

File: tests/ztimer_overhead/Makefile [hunk 2]
```diff
@@ -2,5 +2,7 @@ DEVELHELP ?= 0
 include ../Makefile.tests_common
 
 USEMODULE += ztimer_overhead ztimer_usec
+USEMODULE += ztimer64
+USEMODULE += ztimer64_usec
```
Comment (fjmolinas) [inline tests/ztimer_overhead/Makefile:6]: Unrelated?

**Review by fjmolinas — COMMENTED** [review]
Some comments...

File: sys/net/gnrc/Makefile.dep (hunk 1 above)
Comment (bergzand) [reply sys/net/gnrc/Makefile.dep:138]: The ztimer_usec usage is optional and only used for debugging. I don't want to pull it in by default here.

File: tests/ztimer_overhead/Makefile (hunk 2 above)
Comment (bergzand) [reply tests/ztimer_overhead/Makefile]: Whoops

File: sys/net/gnrc/Makefile.dep (hunk 1 above)
Comment (fjmolinas) [reply sys/net/gnrc/Makefile.dep:138]: Ok

**Comment by fjmolinas** [comment]
Please squash!

**Comment by bergzand** [comment]
> Please squash!

squashed!

**Review by fjmolinas — APPROVED** [review]
ACK, straightforward enough, ran the test on multiple BOARDs all passing.

**Comment by fjmolinas** [comment]
There seems to be an issue, the test does not pass on `frdm-kw41z`, did on master, it simply stalls

**Comment by fjmolinas** [comment]
> There seems to be an issue, the test does not pass on `frdm-kw41z`, did on master, it simply stalls

This issue was fixed with #17393

**Review by fjmolinas — APPROVED** [review]
re-ACK @bergzand can you push a commit adding the required BOARDs to Makefile.ci, since this one is a bit old, could you rebase as well?

File: examples/ccn-lite-relay/Makefile.ci [hunk 3]
```diff
@@ -1,3 +1,4 @@
 BOARD_INSUFFICIENT_MEMORY := \
     stm32f030f4-demo \
+    pba-d-01-kw2x \
```
Comment (fjmolinas) [inline examples/ccn-lite-relay/Makefile.ci:3]: Are you sure this is needed? 

```
BOARD=pba-d-01-kw2x make -C examples/ccn-lite-relay/ -j
Building application "ccn-lite-relay" for "pba-d-01-kw2x" with MCU "kinetis".

"make" -C /home/francisco/workspace/RIOT/pkg/ccn-lite
/home/francisco/workspace/RIOT/dist/tools/cmake/generate-xcompile-toolchain.sh > /home/francisco/workspace/RIOT/build/pkg/ccn-lite/xcompile-toolchain.cmake
cmake -B/home/francisco/workspace/RIOT/examples/ccn-lite-relay/bin/pba-d-01-kw2x/pkg-build/ccn-lite -H/home/francisco/workspace/RIOT/build/pkg/ccn-lite/src \
      -DCMAKE_TOOLCHAIN_FILE=/home/francisco/workspace/RIOT/build/pkg/ccn-lite/xcompile-toolchain.cmake \
      -DCCNL_RIOT=1 -DRIOT_CFLAGS="-isystem /home/francisco/opt/gcc-arm-none-eabi-10-2020-q4-major/arm-none-eabi/include/newlib-nano -I/home/francisco/workspace/RIOT/core/include -I/home/francisco/workspace/RIOT/drivers/include -I/home/francisco/workspace/RIOT/sys/include -I/home/francisco/workspace/RIOT/boards/pba-d-01-kw2x/include -I/home/francisco/workspace/RIOT/cpu/kinetis/include -I/home/francisco/workspace/RIOT/cpu/cortexm_common/include -I/home/francisco/workspace/RIOT/cpu/cortexm_common/include/vendor -I/home/francisco/workspace/RIOT/sys/libc/include -I/home/francisco/workspace/RIOT/build/pkg/ccn-lite/src/ccnl-riot/include -I/home/francisco/workspace/RIOT/build/pkg/ccn-lite/src/ccnl-core/include -I/home/francisco/workspace/RIOT/build/pkg/ccn-lite/src/ccnl-pkt/include -I/home/francisco/workspace/RIOT/build/pkg/ccn-lite/src/ccnl-fwd/include -I/home/francisco/workspace/RIOT/sys/net/gnrc/pktbuf/include -I/home/francisco/workspace/RIOT/sys/net/gnrc/pktbuf_static/include -I/home/francisco/workspace/RIOT/sys/posix/include -I/home/francisco/workspace/RIOT/sys/net/link_layer/eui_provider/include -I/home/francisco/workspace/RIOT/drivers/kw2xrf/include" -DBUILD_TESTING=OFF
Doxygen build started
-- Configuring done
-- Generating done
-- Build files have been written to: /home/francisco/workspace/RIOT/examples/ccn-lite-relay/bin/pba-d-01-kw2x/pkg-build/ccn-lite
make -C /home/francisco/workspace/RIOT/examples/ccn-lite-relay/bin/pba-d-01-kw2x/pkg-build/ccn-lite
[ 11%] Built target ccnl-pkt
[ 13%] Built target ccnl-fwd
[ 48%] Built target ccnl-core
[100%] Built target ccnl-riot
"make" -C /home/francisco/workspace/RIOT/boards/pba-d-01-kw2x
"make" -C /home/francisco/workspace/RIOT/core
"make" -C /home/francisco/workspace/RIOT/cpu/kinetis
"make" -C /home/francisco/workspace/RIOT/drivers
"make" -C /home/francisco/workspace/RIOT/sys
"make" -C /home/francisco/workspace/RIOT/drivers/kw2xrf
"make" -C /home/francisco/workspace/RIOT/drivers/netdev
"make" -C /home/francisco/workspace/RIOT/sys/auto_init
"make" -C /home/francisco/workspace/RIOT/drivers/periph_common
"make" -C /home/francisco/workspace/RIOT/sys/div
"make" -C /home/francisco/workspace/RIOT/cpu/cortexm_common
"make" -C /home/francisco/workspace/RIOT/sys/evtimer
"make" -C /home/francisco/workspace/RIOT/cpu/kinetis/periph
"make" -C /home/francisco/workspace/RIOT/sys/fmt
"make" -C /home/francisco/workspace/RIOT/sys/frac
"make" -C /home/francisco/workspace/RIOT/sys/isrpipe
"make" -C /home/francisco/workspace/RIOT/sys/luid
"make" -C /home/francisco/workspace/RIOT/sys/malloc_thread_safe
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc
"make" -C /home/francisco/workspace/RIOT/sys/net/link_layer/eui_provider
"make" -C /home/francisco/workspace/RIOT/sys/net/link_layer/ieee802154
"make" -C /home/francisco/workspace/RIOT/sys/net/link_layer/l2util
"make" -C /home/francisco/workspace/RIOT/cpu/cortexm_common/periph
"make" -C /home/francisco/workspace/RIOT/sys/net/netif
"make" -C /home/francisco/workspace/RIOT/sys/newlib_syscalls_default
"make" -C /home/francisco/workspace/RIOT/sys/od
"make" -C /home/francisco/workspace/RIOT/sys/pm_layered
"make" -C /home/francisco/workspace/RIOT/sys/ps
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/netapi
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/netif
"make" -C /home/francisco/workspace/RIOT/sys/random
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/netreg
"make" -C /home/francisco/workspace/RIOT/sys/shell
"make" -C /home/francisco/workspace/RIOT/sys/shell/commands
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/pkt
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/pktbuf
"make" -C /home/francisco/workspace/RIOT/sys/stdio_uart
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/pktbuf_static
"make" -C /home/francisco/workspace/RIOT/sys/timex
"make" -C /home/francisco/workspace/RIOT/sys/tsrb
"make" -C /home/francisco/workspace/RIOT/sys/ztimer
"make" -C /home/francisco/workspace/RIOT/sys/ztimer64
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/pktdump
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/netif/hdr
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/netif/ieee802154
"make" -C /home/francisco/workspace/RIOT/sys/net/gnrc/netif/init_devs
   text	   data	    bss	    dec	    hex	filename
  62660	    180	  14848	  77688	  12f78	/home/francisco/workspace/RIOT/examples/ccn-lite-relay/bin/pba-d-01-kw2x/ccn-lite-relay.elf
```

Comment (bergzand) [reply examples/ccn-lite-relay/Makefile.ci]: Oops, fixed with the latests push

**Comment by fjmolinas** [comment]
Thanks @bergzand!
